cmd/src/batch_common.go

```diff
@@ -604,3 +583,14 @@ func getBatchParallelism(ctx context.Context, flag int) (int, error) {
 
 	return docker.NCPU(ctx)
 }
+
+func validateSourcegraphVersionConstraint(ctx context.Context, svc *service.Service) error {
```

Comment:
I worry a little that this may be too simplistic as we start adding things in 4.x again, but let's cross that bridge when we get to it.